pkg/templates/k8sresource_funcs.go

```diff
@@ -60,6 +62,99 @@ func (t *TemplateResolver) fromSecretProtected(namespace string, secretName stri
 	return t.protect(value)
 }
 
+// copies all data in the given Secret, namespace.
+func (t *TemplateResolver) copySecretDataBase(namespace string, secretname string) (map[string]interface{}, error) {
+	klog.V(2).Infof("copySecretDataBase for namespace: %v, secretname: %v", namespace, secretname)
+
+	ns, err := t.getNamespace("copySecretData", namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsClient := (*t.kubeClient).CoreV1().Secrets(ns)
+	secret, getErr := secretsClient.Get(context.TODO(), secretname, metav1.GetOptions{})
+
+	if getErr != nil {
+		if k8serrors.IsNotFound(getErr) {
+			// Add to the referenced objects if the object isn't found since the consumer may want to watch the object
+			// and resolve the templates again once it is present.
+			t.addToReferencedObjects("/v1", "Secret", ns, secretname)
+		}
+
+		klog.Errorf("Error Getting secret:  %v", getErr)
+		err := fmt.Errorf("failed to get the secret %s from %s: %w", secretname, ns, getErr)
+
+		return nil, err
+	}
+
+	data := make(map[string]interface{})
+
+	for key, val := range secret.Data {
+		// when using corev1 secret api, the data is returned decoded,
+		// re-encododing to be able to use it in the referencing secret
+		sEnc := base64.StdEncoding.EncodeToString(val)
+		data[key] = sEnc
+	}
+
+	// add this Secret to list of  Objs referenced by the template
+	t.addToReferencedObjects("/v1", "Secret", secret.Namespace, secret.Name)
+
+	return data, nil
+}
+
+// copies all data in the given Secret, namespace.
+func (t *TemplateResolver) copySecretData(namespace string, secretname string) (string, error) {
+	klog.V(2).Infof("copySecretData for namespace: %v, secretname: %v", namespace, secretname)
+
+	data, err := t.copySecretDataBase(namespace, secretname)
+	if err != nil {
+		return "", err
+	}
+
+	rawData, err := json.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+
+	dst := &bytes.Buffer{}
+	err = json.Compact(dst, rawData)
+
+	if err != nil {
+		return "", err
+	}
+
+	return dst.String(), nil
+}
+
+// copySecretDataProtected wraps copySecretData and encrypts the output value using the "protect" method.
+func (t *TemplateResolver) copySecretDataProtected(namespace string, secretName string) (string, error) {
+	data, err := t.copySecretDataBase(namespace, secretName)
+	if err != nil {
+		return "", err
+	}
+
+	for key, val := range data {
+		data[key], err = t.protect(fmt.Sprint(val))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	rawData, err := json.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+
+	dst := &bytes.Buffer{}
+	err = json.Compact(dst, rawData)
```

Comment:
Is this needed? Is `rawData` not already a single line?

Essentially, we just want the JSON without any new lines in it.

Reply:
I took this out.